agent/engine/dockerstate/docker_task_engine_state.go

```diff
@@ -143,6 +146,22 @@ func (state *DockerTaskEngineState) ContainerMapByArn(arn string) (map[string]*a
 	return ret, ok
 }
 
+// TaskByShortID retrieves the task of a given docker short container id
+func (state *DockerTaskEngineState) TaskByShortID(cid string) ([]*api.Task, bool) {
+	state.lock.RLock()
+	defer state.lock.RUnlock()
+
+	var tasks []*api.Task
+	for id := range state.idToTask {
+		if strings.HasPrefix(id, cid) {
+			task, _ := state.TaskByID(id)
```

Comment:
@aaronwalker thank you! I'm in favor of adding the `AllContainers()` (`GetAllContainerIDs()` in my example code) method to the `TaskEngineState` interface itself (on the lines of `AllTasks()`) and implementing the same in `DockerTaskEngineState`. However, this also probably means that you'd have to update the mock for the interface (by running `make gogenerate`). If you think that's getting too tricky/complicated, I'd happy with a private/local scope method.